.github/workflows/ci-test.yml

```diff
@@ -1,15 +1,53 @@
-name: Test
+name: Test & documentation
 on: [push, pull_request]
+permissions:
+  contents: read
+env:
+  MAVEN_OPTS: -XX:StartFlightRecording=maxsize=5g,disk=true,dumponexit=true,settings=default,filename=./ -DtrimStackTrace=false -D'maven.resolver.transport=wagon'
+  DEV_JDK: '17'
 jobs:
+  license:
+    name: License check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: ${{ env.DEV_JDK }}
+          cache: 'maven'
+      - run: mvn -V -B license:check
+        timeout-minutes: 60
+  dependencies:
+    name: Dependency checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: ${{ env.DEV_JDK }}
+          cache: 'maven'
+      - run: mvn -V -B dependency-check:check
+        timeout-minutes: 60
+  documentation:
+    name: Javadoc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: ${{ env.DEV_JDK }}
+          cache: 'maven'
+      - run: mvn -V -B -q -T 2C install javadoc:javadoc -DskipTests -D'dependency-check.skip' -D'license.skip' --projects '!exist-distribution,!exist-installer' --also-make
```

Comment:
I am not sure if it makes sense to have the javadoc in a separate task.
As it is setup in this PR, the javadoc CI job will also call these Maven phases:

1. validate
2. initialize
3. generate-sources
4. process-sources
5. generate-resources
6. process-resources
7. compile
8. process-classes 
9. generate-test-sources
10. process-test-sources
11. generate-test-resources
12. process-test-resources
13. test-compile
14. process-test-classes
15. test
16. prepare-package
17. package
18. pre-integration-test
19. integration-test
20. post-integration-test
21. verify
22. install

Only phases 15 and 19, i.e. `test` and `integration test` will be reduced (but not omitted) in this job due to the `-DskipTests` short-cutting the `process` methods of the `maven-surefire-plugin` and `maven-failsafe-plugin`. As all of these phasse are already performed in the `Maven Test` CI job, separating this into a separate job has increased the amount of work required by the CI due to duplication of phases.

I think it would be better to reinstate it into the `Maven Test` CI job with a condition that it is only executed on the Linux runner (as that is the fastest runner).

Reply:
would this mean no unit tests anymore on windows/macos?

Reply:
No, it would mean that only the `javadoc` task would be run on Linux.